vm/systemSmartContracts/validator.go

```diff
@@ -121,6 +128,15 @@ func NewValidatorSmartContract(
 	if len(args.GovernanceSCAddress) < 1 {
 		return nil, fmt.Errorf("%w for governance sc address", vm.ErrInvalidAddress)
 	}
+	if check.IfNil(args.NodesCoordinator) {
+		return nil, fmt.Errorf("%w in validatorSC", vm.ErrNilNodesCoordinator)
+	}
+	if args.StakingSCConfig.NodeLimitPercentage < minPercentage {
+		return nil, fmt.Errorf("%w in validatorSC", vm.ErrInvalidNodeLimitPercentage)
```

Comment:
`fmt.Errorf("%w in validatorSC for NodeLimitPercentage", vm.ErrInvalidNodeLimitPercentage)`

Reply:
no. as you would duplicate the node limit percentage write

Reply:
yeah, now I've saw ErrInvalidNodeLimitPercentage vs ErrInvalidStakeLimitPercentage